app/main.py

```diff
@@ -2,24 +2,64 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount: int = min_amount
+        self.max_amount: int = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.protected_name: str = "_" + name
+
+    def __get__(self, instance: object, owner: type) -> int | None:
+        return getattr(instance, self.protected_name, None)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError(f"Value {value} is out of range "
+                             f"({self.min_amount}-{self.max_amount})")
+        setattr(instance, self.protected_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name: str = name
+        self.age: int = age
+        self.weight: int = weight
+        self.height: int = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age: int = age
+        self.weight: int = weight
+        self.height: int = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
 
 
 class Slide:
-    pass
+    def __init__(
+            self, name: str, limitation_class: type[SlideLimitationValidator]
```

Comment:
The type hint for `limitation_class` should be `type` instead of `type[SlideLimitationValidator]` because `limitation_class` is expected to be a class, not an instance. Using `type` is more appropriate for class references.